src/lib/incentive-relationship-calculation.ts

```diff
@@ -149,6 +150,7 @@ export function makeIneligible(incentiveId: string, maps: RelationshipMaps) {
     for (const dependentId of maps.supersedesMap.get(incentiveId)!) {
       if (
         maps.ineligibleIncentives.has(dependentId) &&
+        !maps.permanentlyIneligibleIncentives.has(dependentId) &&
```

Comment:
Optional: move the check into MakeEligible itself and return early if the incentive is permanently ineligible. That goes back to the question of what's the responsibility of the caller vs the function itself, but in this case I think a function called MakeEligible ought to know about permanent ineligibility. I also think the list of things we expect all the callers to check is getting a little long and I'm worried about someone eventually missing this in the future.

I won't block on it though, so if you have strong opinions that it's better outside, that's fine.

Reply:
Yeah I totally agree on this one, it's better inside makeEligible. I'm making the switch.